File: JavaScript/js/account.js


function Account (owner, accNum, balance) {         // 파스칼표기 1
    this.owner = owner;
    this.accNum = accNum;
    this.balance = balance;
    this.name = "직장인 비과세 예금";
    this.bankName = "우리은행";

    // console.log(this);

    // 객체 기능 부여 (메서드) 3
    this.checkBalnace = function() {
        return this.balance;
    }
    this.deposit = function(money) {
        this.balance += money;
    };

    this.withdraw = function(money) {
        if (this.balance >= money) {

            this.balance -= money;
        } else {
            window.alert("잔액이 부족합니다.");
        }
    }
}

File: java/OOPEx1_Constructor/src/webkdh/Account.java
package webkdh;

public class Account {
	private int balance = 0; // 계좌 잔액
	int deposit; // 입금
	int withdraw; // 출금
	int getbalance; // 현재 금액
	
	// 입금 메서드
	public void deposit(int amount) {
		if (isValid(amount)) {
			balance += amount;
			System.out.println(amount + "원이 입금되었습니다.");
		} else {
			System.out.println("유효하지 않은 금액입니다.");
		}
	}
	// 출금 메서드
	public void withdraw(int amount) {
		if (isValid(amount) && (this.balance - amount) >= 0) {
			balance -= amount;
				System.out.println(amount + "원이 출금되었습니다.");
			} else {
				System.out.println("유효하지 않은 금액이거나 잔액이 부족합니다.");
			}
	}
	
	// 잔액 조회 메서드
	public int getBalance() {
		return balance;
	} 
	
    // 금액이 유효한지 검사하는 private 메서드
    private boolean isValid(int amount) {
        return amount > 0;
    }
}
